# HG changeset patch
# User fyang
# Date 1496998140 -28800
#      Fri Jun 09 16:49:00 2017 +0800
# Node ID 125269abe0e7a552ae4bddcef82993c153b0364e
# Parent  b139704a6ceef1503f24283f93fb6e87060526a6
7009641, PR3423: Don't fail VM when CodeCache is full
Summary: Port 7009641 to AArch64
Reviewed-by: aph
Contributed-by: teng.lu@linaro.org

diff --git a/src/cpu/aarch64/vm/vtableStubs_aarch64.cpp b/src/cpu/aarch64/vm/vtableStubs_aarch64.cpp
--- openjdk/hotspot/src/cpu/aarch64/vm/vtableStubs_aarch64.cpp
+++ openjdk/hotspot/src/cpu/aarch64/vm/vtableStubs_aarch64.cpp
@@ -52,6 +52,11 @@
 VtableStub* VtableStubs::create_vtable_stub(int vtable_index) {
   const int aarch64_code_length = VtableStub::pd_code_size_limit(true);
   VtableStub* s = new(aarch64_code_length) VtableStub(true, vtable_index);
+  // Can be NULL if there is no free space in the code cache.
+  if (s == NULL) {
+    return NULL;
+  }
+
   ResourceMark rm;
   CodeBuffer cb(s->entry_point(), aarch64_code_length);
   MacroAssembler* masm = new MacroAssembler(&cb);
